src/core/Form/MultiSelect/MultiSelect/MultiSelect.tsx

```diff
@@ -680,6 +719,25 @@ class BaseMultiSelect<T> extends Component<
           {selectedItems.length}
           {ariaSelectedAmountText}
         </VisuallyHidden>
+        <VisuallyHidden
+          aria-live="polite"
+          aria-atomic="true"
+          id={`${id}-filteredItems-length`}
+        >
+          {this.focusInInput(this.getOwnerDocument()) ? (
+            <>
+              {filteredItems.length}
+              {ariaOptionsAvailableText}
+            </>
+          ) : null}
```

Comment:
Safari doesn't seem to work nicely with aria region when I first tried like this (might had some error):

```tsx
{this.focusInInput()
  ? `${filteredItems.length} ${ariaFilteredAmountText}`
  : null}
```